public/app/plugins/datasource/tempo/language_provider.ts

```diff
@@ -36,24 +39,58 @@ export default class TempoLanguageProvider extends LanguageProvider {
     return res?.data;
   };
 
-  start = async () => {
-    if (!this.startTask) {
-      this.startTask = this.fetchTags().then(() => {
+  start = async (range?: TimeRange, timeRangeForTags?: number) => {
+    // Check if we need to refetch tags due to range changes (minute-level granularity)
+    const shouldRefetch = this.shouldRefreshLabels(range, this.previousRange);
+
+    if (!this.startTask || shouldRefetch) {
```

Comment:
Could this cause issues if the previous request is still pending? Do we cancel ongoing requests?

Reply:
Testing it out it does not seem to cause issue, just creates another request if the range changes before the other one returns. I don't think Tempo allows us to cancel requests.